components/measurement/CommonDetails.js

```diff
@@ -0,0 +1,124 @@
+/* global require */
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import {
+  Container,
+  Heading,
+  Border,
+  Button,
+  Flex,
+  Box,
+  Text,
+  theme
+} from 'ooni-components'
+
+import MdFileDownload from 'react-icons/lib/md/file-download'
+import NoSSR from 'react-no-ssr'
+
+// We wrap the json viewer so that we can render it only in client side rendering
+class JsonViewer extends React.Component {
+  render() {
+    const ReactJson = require('react-json-view').default
+    const {
+      src
+    } = this.props
+    return (
+      <ReactJson src={src} />
+    )
+  }
+}
+
+JsonViewer.propTypes = {
+  src: PropTypes.object.isRequired
+}
+
+const softwareNames = {
+  'ooniprobe-ios': {
+    software: 'OONI Probe',
+    platform: 'iOS'
+  },
+  'ooniprobe-android': {
+    software: 'OONI Probe',
+    platform: 'Android'
+  }
+}
```

Comment:
I think it's more robust to get the target platform from the `annotations->platform` key. This way we can get it using the same strategy for `ios`, `macos`, `android`, `windows`, `linux` and `lepidopter`.

Reply:
See the base data format spec WRT to the `annotations` key: https://github.com/ooni/spec/blob/master/data-formats/df-000-base.md#specification